### PR TITLE
Remove LINQ errors for filtering

### LINKED ISSUE #1229: Fix filtering section on history page
Describe the bug
Flotilla crashes when a letter is written in the "Robot name" or "Tag" search field of the filtering section.

To Reproduce
Go to the history page and enter a letter in one of the search fields.

Expected behavior
Should not crash, but filter out the irrelevant information.

Image

### PR DESCRIPTION
**Pull request by andchiind**
Closes #1229

### CODE REVIEW

File: backend/api/Services/MissionRunService.cs [hunk 1]
```diff
@@ -198,7 +198,7 @@ private static void SearchByName(ref IQueryable<MissionRun> missionRuns, string?
 
             missionRuns = missionRuns.Where(
                 missionRun =>
-                    missionRun.Name != null && missionRun.Name.Contains(name.Trim(), StringComparison.OrdinalIgnoreCase)
+                    missionRun.Name != null && missionRun.Name.Contains(name.Trim())
```
Comment (prasm313) [inline backend/api/Services/MissionRunService.cs:201]:
```suggestion
                    missionRun.Name != null && missionRun.Name.ToLower().Contains(name.ToLower().Trim())
```

File: backend/api/Services/MissionRunService.cs [hunk 2]
```diff
@@ -207,7 +207,7 @@ private static void SearchByRobotName(ref IQueryable<MissionRun> missionRuns, st
             if (!missionRuns.Any() || string.IsNullOrWhiteSpace(robotName)) { return; }
 
             missionRuns = missionRuns.Where(
-                missionRun => missionRun.Robot.Name.Contains(robotName.Trim(), StringComparison.OrdinalIgnoreCase)
+                missionRun => missionRun.Robot.Name.Contains(robotName.Trim())
```
Comment (prasm313) [inline backend/api/Services/MissionRunService.cs:210]:
```suggestion
                missionRun => missionRun.Robot.Name.ToLower().Contains(robotName.ToLower().Trim())
```

File: backend/api/Services/MissionRunService.cs [hunk 3]
```diff
@@ -219,8 +219,10 @@ private static void SearchByTag(ref IQueryable<MissionRun> missionRuns, string?
                 missionRun =>
                     missionRun.Tasks.Any(
                         task =>
+#pragma warning disable CA1307
                             task.TagId != null
-                            && task.TagId.Contains(tag.Trim(), StringComparison.OrdinalIgnoreCase)
+                            && task.TagId.Contains(tag.Trim())
```
Comment (prasm313) [inline backend/api/Services/MissionRunService.cs:224]:
```suggestion
                            && task.TagId.ToLower().Contains(tag.ToLower().Trim())
```

File: backend/api/Services/MissionRunService.cs (hunk 1 above)
Comment (prasm313) [reply backend/api/Services/MissionRunService.cs:201]: Suggestion: Make sure that the robots and tags appear regardless of letter case.

File: backend/api/Services/MissionRunService.cs (hunk 3 above)
Comment (andchiind) [reply backend/api/Services/MissionRunService.cs:224]: This is a guid or integer I think so I don't think changing the case would make sense

**Review by prasm313 — APPROVED** [review]
LGTM

**Pull request merged**